src/Windows/R/Editor/Impl/SuggestedActions/RSuggestedActionsSource.cs

```diff
@@ -88,20 +82,26 @@ internal sealed class RSuggestedActionsSource : SuggestedActionsSourceBase, ISug
                     .Select(ap => new SuggestedActionSet(ap.GetSuggestedActions(TextView, TextBuffer, bufferPosition), applicableToSpan: applicableSpan));
         }
 
-        public Task<bool> HasSuggestedActionsAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken) {
-            if (TextView != null && !TextView.Caret.InVirtualSpace) {
-                var rPosition = TextView.MapDownToR(TextView.Caret.Position.BufferPosition);
-                if (rPosition.HasValue && SuggestedActionProviders.Any(actionProvider => actionProvider.HasSuggestedActions(TextView, TextBuffer, rPosition.Value.Position))) {
-                    return Task.FromResult(true);
-                }
-            }
-            return Task.FromResult(false);
-        }
+        public Task<bool> HasSuggestedActionsAsync(ISuggestedActionCategorySet requestedActionCategories, SnapshotSpan range, CancellationToken cancellationToken)
+            => Task.FromResult(
+                    TextView != null && 
+                    SuggestedActionProviders.Any(a => a.HasSuggestedActions(TextView, TextBuffer, PositionFromRange(range))));
 
         public bool TryGetTelemetryId(out Guid telemetryId) {
             telemetryId = REditorCommands.REditorCmdSetGuid;
             return true;
         }
         #endregion
+
+        /// <remarks>
+        /// Picks position in range to fetch suggested acions. 
+        /// Ideally range content should be split and analyzed 
+        /// but this is TODO in the future.
+        /// </remarks>
+        private static int PositionFromRange(SnapshotSpan range) {
+            var text = range.Snapshot.GetText(range);
+            var nonWsOffset = text.Length - text.TrimStart().Length;
```

Comment:
Looks like an overkill to create a new string just to find its index, but since there is a TODO, let's leave it as it is.